app/controllers/UserTypeController.scala

```diff
@@ -41,7 +42,20 @@ class UserTypeController @Inject()(identify: IdentifierAction,
                                    view: UserTypeView)
   extends FrontendController(mcc) with I18nSupport {
 
-  private lazy val backLink: Call = controllers.routes.ConfirmEORIDetailsController.onLoad()
+  private[controllers] def backLink()(implicit request: OptionalDataRequest[AnyContent]): Call = {
+    val cyaMode = {
+      for {
+        answers <- request.userAnswers
+        mode <- answers.get(CheckModePage)
+      } yield mode
+    }.getOrElse(false)
+
+    if (cyaMode) {
+      controllers.routes.CheckYourAnswersController.onLoad()
+    } else {
+      controllers.routes.ConfirmEORIDetailsController.onLoad()
+    }
+  }
```

Comment:
We would normally do the back links in the below format. Is there is reason to why you done the back link like you have? 

 private[controllers] def backLink()(implicit request: DataRequest[_]): Call = {
    if (request.checkMode) {
      controllers.routes.CheckYourAnswersController.onLoad()
    } else {
      controllers.routes.ConfirmEORIDetailsController.onLoad()
    }
  }

Reply:
Yes, typical controllers include the `DataRequiredAction` that attaches certain functions to the `request` such as `request.checkMode`. However, the `UserTypeController` only has the `DataRetrievalAction` which doesn't have the associated helper functions so I've replicated the functiobnality for this instance.